### DISCUSSION TITLE
Erroneous GitHub warning messages about Multi-Factor Authentication

### DISCUSSION THREAD

**Discussion by cosmic-linden** [community/community#129244] [Code Security] [answered]
### Select Topic Area

Bug

### Body

GitHub has begun showing me a yellow warning message which says:

> Please configure another 2FA method to reduce your risk of permanent account lockout. We strongly recommend against SMS as it is prone to fraud and delivery may be unreliable depending on your region.

In addition, I see the following warning message in the Password and authentication tab:

> Please configure another 2FA method to reduce your risk of permanent account lockout. We strongly recommend against SMS as it is prone to fraud and delivery may be unreliable depending on your region. 

Contrary to the above messages, I do not have SMS-based authentication enabled for my account, let alone as the sole second-factor authentication method. First, SMS-based authentication is disabled. Second, I have code-based MFA enabled, which is a widely-supported open standard, as well as the security codes provided by GitHub.

Taken together, this security message is misleading, and may lead to some users becoming confused and inadvertently weakening their security practices. It is also possible that there could be a bug here, causing the message to not be shown to users who have SMS-based authentication enabled.

**Comment by SampsonCrowley** [comment]
This message is really pissing me off. I should not have to constantly see a misleading warning about not having enough 2FA methods. I have ONE and I only want ONE. I do NOT want there to ever be ANY other method that a hacker can use to bypass my 2FA. there is ZERO chance that I will lose access to my authenticator keys. and if I did I'd also have to have permanently lost access to my bitwarden password manager which at that point, GitHub is the least of my worries for the things that I will never be able to access again.

GitHub: Don't tell me I need to lessen my security. I chose a method, and only that method, fully aware of the implications involved if I were to ever be stupid enough to lose those values.

**Reply by SampsonCrowley** [reply]
At least make the message go away forever when I dismiss it...

**Comment by davevad93** [comment]
Hi @cosmic-linden ,

Yes, the warning statement can be misleading because it actually refers that you should have more than one 2FA method enabled and avoid SMS authentication. The warning goes away if you add another 2FA method (even SMS), I've tested it yesterday.

See this discussion #129189

**Reply by poshcodebear** [reply]
But this makes no sense if I've configured recovery codes already, which *is* in effect an additional (one-time-use, break-glass) 2FA method keeping me from getting locked out of my account

**Reply by prahladyeri** [reply]
@poshcodebear The recovery codes don't count as "Preferred 2FA method" which you must set in the settings.
They basically want you to use an Authenticator app such as Google Authenticator or Twilio Authy to generate a TOTP. The SMS way is becoming obsolete and less trusted it seems.

**Reply by poshcodebear** [reply]
Except I am using an authenticator app, and I have recovery codes, and I'm not using SMS... So the warning makes zero sense

**Reply by brill-touchbistro** [reply]
I'm already using passkeys, you can't tell me that isn't a better way.

**Comment by mark-henry01** [comment]
Three-factor authentication? It is getting ridiculous.

Surely this must be a bug in the notification system, given two-factor authentication is already enabled by usage of Authenticator app (and not SMS) .

**Reply by prahladyeri** [reply]
If you already use an Authenticator app, try to set it as "Preferred 2FA method" in your settings, that might help.

**Reply by Bob837217** [reply]
I get this message as well.  Annoys the hell out of me.  I see in my settings that there are two "Add" buttons present.  One for SMS (ooh SMS bad though) and one for "Github Mobile".  Is this a subtle way of steering users into adding "Github Mobile"?  Because I don't want to do that.  And as others have stated - their existing configurations and downloaded break glass codes will ensure no lockout event happens.  Is this the obligatory "enshitification phase" of Github, post MS buyout?

**Reply by opyate** [reply]
> Three-factor authentication? It is getting ridiculous

Exactly, they need to change the wording from "Please configure another 2FA method" to "Please configure 3FA" or "Please configure a third factor", because "another 2FA" would bring me up to 4FA. It's just maths, innit.

**Reply by Dawoodoz** [reply]
Soon it will be "you may only use Microsoft Authenticator with a default password on Microsoft Windows, for your uuuuuuh... superior security when logging into our Linux servers.".

If they really worry about people being locked out, they should just ask for passport or social security numbers and let people regain access in person with biometric checks for an administrative fee of $50.

**Reply by joshi1983** [reply]
@Dawoodoz there are enough data breaches involving social insurance numbers that the number would be a horrible way to authenticate someone.  The number also isn't tied to your photo so an imposter would just need to discover your number to pass that aspect of authentication.  You have to give it to a huge number of people including many government agencies, employers, banks, credit unions, insurance companies, and many databases will keep it.  Even if you don't know it was compromised, it most likely was.  A passport and its photo matching someone's face and the face stored with the account after would be better.  Data breaches and the prevalence of identity theft make it very difficult to authenticate anyone.

Sharing a photo of your face with github could be helpful but sharing your social insurance number or passport number would be a serious invasion of privacy and put more of your identity at risk.  If you met in person with a passport, driving license or government issued photo ID, they could visually match your face and your passport's photo and compare the names on that ID with the real names in the github account.  They could also perform some security checks on the passport and possibly download the associated photo from a government passport database so a fake passport would be discovered.  Having prior knowledge of your passport number would be far less helpful than the risk to your identity because anything you expect to happen in person should be with your real government ID anyway.  Maintaining a copy of those numbers in github won't help much but would be an extra place where that data could be breached.

**Reply by Bob837217** [reply]
@Daoodoz @joshi1983 

Joshi1983??  Add 1 onto that number and you get George Orwell.
Facebook already does this: "[Why Facebook may ask you to upload an ID | Facebook Help Center](https://www.facebook.com/help/314201258613998)"

Please do not encourage this bad behaviour.  I won't give any identity document to any big tech.  That is incredibly naive and a huge persistent risk of identity theft for you.  What happens when that big tech service gets hacked?  When.. not if.  Also, why do they need your passport photo page?  Are they going to verify that with the issuing PassPort Authority in Tajikistan?  No?  Didn't think so.

**Reply by Dawoodoz** [reply]
You don't store it in plain text, only a hash.

**Reply by joshi1983** [reply]
@Bob837217 I typed "Sharing a photo of your face with github could be helpful".  I didn't say a photo of your passport page.  Did you even read my comment?  Why did you misread that as a photo of the passport page?  I was discouraging sharing of the passport number.  I didn't encourage sharing of anything from the passport.  I was warning about the same data breach risk you were.

A photo of your face could even be a publicly shared photo that github just knows their authenticated user indicated looks like him/her.  It is easy to get a photo of you but it can be difficult for an imposter to make their face look like yours if they don't happen to already look a lot like you.  Unless you have never talked with a news reporter, kept your photo off your employer's website, keep all your social media private, and all your close friends and family keep their photos with you private, your face is likely a fairly quick search away.

I mentioned that they might be able to connect with an official passport database as a way to make sure the passport isn't fake.  If they can't do that, they could likely do some checks on the government ID that border security and government agencies do to verify that it isn't fake.  The photo of the github user's face compared with the face of the person seeing them in person would mostly authenticate the user regardless of the official government photo ID.  The government photo ID shown only in person would just be extra evidence that the names from the github account match the face of the person they're face to face with.

**Comment by deloz** [comment]
![image](https://github.com/community/community/assets/685167/331c32de-b9ee-4cf5-90cc-5532495f96b5)

So what do I need to do?

**Reply by Bob837217** [reply]
Nothing at all.  Looks like a "scare-o-gram" to get you to sign up additionally to GitHub Mobile.  Just ignore it.

**Reply by SkyeCyclone** [reply]
Feitian

**Reply by SkyeCyclone** [reply]
> Feitian
> 

FIDO IS BEST FOR CYBER SECURITY AWARENESS COMPANIES

**Comment by joshi1983** [comment]
This message shows more often on github to people who have 2FA already set up than the Jimmy Wales donate message on Wikipedia.  That says a lot.

**Reply by Bob837217** [reply]
Well, the line has to go up, doesn't it?  Probably need some uBlock Origin rules set up for GitHub.com now to remove offending notifications.

**Comment by mjmatthiesen** [comment]
How can I remove this warning, permanently? It's junk. I use an auth app, I am not configuring SMS, and the other two options aren't viable for a sane person. So I have my backup codes as the second.

**This warning should not be shown and is irresponsible to show it**.

**Reply by Bob837217** [reply]
You best option is to use uBlock Origin picker and create a rule based on that input.

**Reply by mjmatthiesen** [reply]
I would certainly argue that Github should not be displaying erroneous security messages and allow us to dismiss them without resorting to third party tools.

**Reply by Bob837217** [reply]
Yep but MS has to make money for shareholders and I guess they want to coerce users into installing a data harvesting app?

**Reply by LightAndLight** [reply]
> I use an auth app, I am not configuring SMS, and the other two options aren't viable for a sane person [...] I have my backup codes as the second.

I'm in the same situation, which is why I thought the warning was weird, then searched for it and ended up here. :face_with_diagonal_mouth: 

I'm blocking the banner using this uBlock Origin query: `##inline-security-checkup-notice > x-banner > .inline-two-factor-methods-banner`.

**Reply by SkyeCyclone** [reply]
Oohh

**Comment by SkyeCyclone** [comment]
Fido key

**Comment by giuliano-macedo** [comment]
I love that it says "permanent account lockout" without any deadline and says "reduce your risk" as if MS is going to roll a dice to decide which dev is going to be locked out for no reason.

**Reply by Bob837217** [reply]
That's a nice github account you have there, would be a shame if something were to suddenly happen to it?

**Reply by giuliano-macedo** [reply]
![image](https://github.com/community/community/assets/13181797/0f6e9c3d-be09-4789-a157-d50c90834aa2)

**Reply by SkyeCyclone** [reply]
What is that supposed to mean why is it at risk?

**Reply by Dawoodoz** [reply]
A phone number can be registered in one's identity and then recovered if the SIM card is lost, so then you just need fireproof storage of passwords.

**Comment by lcdunne** [comment]


**Reply by davevad93** [reply]
Hey @lcdunne , don't worry you'll not be permanently locked out, the warning is a bit misleading, let's say it's just an "advice" to use more than one `2FA` method and avoid `SMS` auth because it's not safe. The funny thing is that if you add SMS authentication as a second (or third) `2FA` method, the warning goes away even though they say that it should be avoided for security reasons. 🤣

**Comment by N-Usha** [comment, marked as answer]
Thanks for all your feedback folks and apologies for the confusion and inconvenience caused.

As a quick update to our [recently shipped security checkup experience](https://github.blog/changelog/2024-06-20-2fa-security-checkup-experience-is-now-improved/), we have now rolled out a patch to restrict the new security check banners from being shown to users who already have a more secure 2FA method configured - Eg: Users who have configured an authenticator app.

Which would mean that starting today, **we would only prompt the users who have configured SMS as their only 2FA method** to review and update their settings, since they need to have more than one 2FA method with a more secure option like an authenticator App. 

We hope that this change would help us influence more developers who contribute code on GitHub.com to enable one or more forms of secure two-factor authentication (2FA).

**Reply by giuliano-macedo** [reply]
Makes a lot of sense, thanks for the update @N-Usha !

**Reply by Sa00840** [reply]
G

**Reply by Ravi-Quantaris** [reply]
It doesn't seem to be working for organizations under specific circumstances... not exactly sure which ones, but I have submitted a ticket highlighting this issue as I have **all** 2FA settings set while still being greeted by the same message whenever I try to access the organization. 

Not sure if other users are also facing this issue when it comes to organizations.

**Reply by brill-touchbistro** [reply]
It's not working, I don't have SMS set up at all (I agree, its not secure).
I'm already using Passkeys, but I keep getting harassed about setting up 2FA.

**Comment by winer328** [comment]
![image](https://github.com/user-attachments/assets/6563f98a-2217-495b-84ae-0ed22b9caa89)
During configure 2FA authentication, I got this error.
How can I solve this error?

**Comment by Alirezaseyed** [comment]
Here’s how you can guide them without a formal greeting:

---

### Steps to Resolve the Erroneous MFA Warning:

1. **Check Your 2FA Settings:**
   - Head to **Settings > Password and Authentication** on your GitHub account.
   - Verify that **SMS-based 2FA** is indeed disabled, and ensure that your **code-based MFA** (like TOTP) is correctly set up. Also, double-check that you have a backup method (like security codes or keys) configured.

2. **Clear Cache and Cookies:**
   - Sometimes, old data stored in your browser can cause false warnings. Try clearing your browser’s cache and cookies, then log back into GitHub to see if the warning persists.

3. **Test with a Different Browser or Incognito Mode:**
   - If the warning is still there, try accessing GitHub from a different browser or in **Incognito Mode** to rule out any browser-related issues.

4. **Report the Issue to GitHub Support:**
   - If none of these steps resolve the problem, consider submitting a bug report to GitHub. In the report, include:
     - A screenshot of the warning message you're seeing.
     - A description of your current 2FA setup (e.g., SMS disabled, using code-based MFA).
     - The approximate time when the warning first appeared.
   
5. **Optional: Check GitHub’s Issue Tracker:**
   - Sometimes, these issues affect multiple users. You can check GitHub’s **issue tracker** for similar reports. If there’s an ongoing issue, it might already be under investigation.

---

Following these steps should help you figure out if the warning is a bug or just a display issue. If needed, reporting it directly to GitHub should lead to a resolution.
